### PR TITLE
group ~> group

### PR DESCRIPTION
**Pull request by tranma**
just the GroupBy first

! @amosr

### CODE REVIEW

File: icicle-source/src/Icicle/Source/Checker/Invariants.hs [hunk 1]
```diff
@@ -77,12 +74,12 @@ invariantQ ctx (Query (c:cs) xfinal)
...
   goBanAll
      = invariantQ
-        (ctx{ checkInvariants = inv{ allowLatest = False, allowWindowsOrGroups = False }})
+        (ctx{ checkInvariants = inv{ allowLatest = False, allowWindowsOrGroups = True }})
         q'
 
   goBanGroup
      = invariantQ
-        (ctx{ checkInvariants = inv{ allowWindowsOrGroups = False }})
+        (ctx{ checkInvariants = inv{ allowWindowsOrGroups = True }})
```
Comment (amosr) [inline icicle-source/src/Icicle/Source/Checker/Invariants.hs]: is `allowWindowsOrGroups` still necessary then? maybe these should remain `False` and just rename it to `allowWindows`

File: icicle-source/src/Icicle/Source/ToCore/Fold.hs [hunk 2]
```diff
@@ -297,11 +303,76 @@ convertFold q
...
+                 , CE.xVar n'x'
+                 , CE.xVar n'k'
+                 , CE.xVar n'r' ]
+
+           let xtra
+                = CE.xPrim (C.PrimFold (C.PrimFoldMap t'k t'fold) t'sum)
+                  CE.@~ ins
+                  CE.@~ x'sumzero
+
+           return $ ConvertFoldResult kons zero xtra t'map t'sum
```
Comment (amosr) [inline icicle-source/src/Icicle/Source/ToCore/Fold.hs]: yep

**Comment by amosr** [comment]
looks good. can you add group-inside-group to the arbitraries, or will that happen because of the type checker changes?

**Comment by tranma** [comment]
@amosr that's already happening yeah

**Comment by tranma** [comment]
this is good enough for nested groups and distincts

**Comment by amosr** [comment]
oh yeah I meant 👍

**Pull request merged**